### ISSUE TITLE
cat() using negative dimension

### ISSUE THREAD

**Issue by wddabc** [pytorch/pytorch#1213]
Currently it gets the same error as #657

**Comment by soumith** [comment]
what is your pytorch version?

```
print(pytorch.__version__)
```

**Comment by wddabc** [comment]
Thanks for the fast reply! It is 0.1.11_5

**Comment by wddabc** [comment]
Oh, sorry for the insufficient information! The difference between this error and #657 is it occurs during backward().
The error come from `https://github.com/pytorch/pytorch/blob/master/torch/autograd/_functions/tensor.py#L313`
, where `self.dim` is negative in `grad_output.narrow(self.dim, end - size, size)`

**Comment by soumith** [comment]
thanks. Can you please see if you can generate this error with a minimal example in say 10 lines of code?
I want to run this and debug the error.

**Comment by wddabc** [comment]
Sure
```
import torch
from torch.autograd import Variable
x = Variable(torch.Tensor([[1,0],[0,1]]), requires_grad=True)
xx = torch.cat([x,x], -1) #crashes
#xx = torch.cat([x,x], 1) #works
obj = torch.sum(xx)
obj.backward()
```

**Comment by albanD** [comment]
I cannot reproduce in the master branch, this issue has been fixed by #1108

**Comment by soumith** [comment]
great. should be fixed in the next release.
